### ISSUE TITLE
restic should always respect DEBUG_LOG without special debug build

### ISSUE THREAD

**Issue by tecknicaltom** [restic/restic#1842]
## Output of `restic version`
restic 0.9.0 compiled with go1.10.1 on linux/amd64

## How did you run restic exactly?
export RESTIC_REPOSITORY="s3:https://s3.amazonaws.com/..."
export RESTIC_PASSWORD="..."
export AWS_PROFILE=restic
DEBUG_LOG=/tmp/restic-debug.log restic init

## What backend/server/service did you use to store the repository?
N/A

## Expected behavior
Specifying DEBUG_LOG (or some other cli flag or environment variable, if desired) should enable debugging without the need for a special build, so that debug logging can be used to help diagnose usage problems.

## Actual behavior
The file referenced by DEBUG_LOG is not touched. As indicated by https://restic.readthedocs.io/en/latest/090_participating.html?highlight=debugging#debugging a special debug build is required to get debug logging.

## Steps to reproduce the behavior
run restic with DEBUG_LOG set to a file path

## Do you have any idea what may have caused this?
debug logging is a build-time option

## Do you have an idea how to solve the issue?
make debug logging always an option, without needing a special debug build

## Did restic help you or made you happy in any way?
hopefully soon!

**Comment by fd0** [comment]
Thanks for taking the time to report this. It's not so easy as to just build every binary with the debug tag, because these binaries are 10-25% slower.

Would you mind elaborating a bit on your use case? Is it too much to rebuild the binary?

**Comment by tecknicaltom** [comment]
I've had problems using restic in two different environments. The one that I ran into last night was fixed by upgrading (related to S3 permissions, I believe fixed by https://github.com/restic/restic/pull/1648) and a previous failure in another environment was due to an sftp server misbehaving. Regardless of the underlying cause of my problems, debuging the issues was more difficult than it should have been. I installed restic, as I imagine most people do, through my distro's package management. This leaves me with a binary that has no means of getting more info out of. Most programs support some type of --verbose or --debug cli flag to get more detailed output. When restic fails, your only recourse is to rebuild the program from source since the program isn't giving you anything more info...

I'm not going to pretend to know everything that's enabled with the debug build that would result in a 10-25% slowdown, and I'm surely not suggesting that all builds should be built with full debugging symbols, etc (I'm not a go developer), but some way to get more debug tracing and errors from the binary when things go wrong would help a lot with troubleshooting.
